api/admin/controller.py

```diff
@@ -212,8 +212,11 @@ def authenticated_admin(self, admin_details):
         )
         if is_new and admin_details.get("roles"):
             for role in admin_details.get("roles"):
-                library = Library.lookup(self._db, role.get("library"))
-                admin.add_role(role.get("role"), library)
+                if role.get("role") in AdminRole.ROLES:
+                    library = Library.lookup(self._db, role.get("library"))
+                    admin.add_role(role.get("role"), library)
```

Comment:
If the lookup fails, `library` will end up `None` and `add_role` will create the role globally, which I don't think we want.